indico/htdocs/js/indico/Management/CFA.js

```diff
@@ -733,16 +733,23 @@ type("AbstractFilesManager", [],
 
     _buildExistingElement: function(pos) {
         var file = this.initialMaterial[pos];
-        var div = Html.div({id:'divExFile_' + pos, className:'existingAttachment'});
+        var div = Html.div({id: 'divExFile_' + pos, className: 'existingAttachment'});
         var a = Html.a({href: file['url']}, file['file']['fileName']);
+        var fileId = this._extractFileIdFromUrl(file['url']);
         div.append(a);
         var imageRemove = this._getImageRemove(pos);
         div.append(imageRemove);
-        var inputHidden = Html.input('hidden', {name: 'existingFile'}, file['id']);
+        console.debug(file);
+        var inputHidden = Html.input('hidden', {name: 'existingFile'}, fileId);
         div.append(inputHidden);
         return div;
     },
 
+    _extractFileIdFromUrl: function(file) {
+        var filename = file.split('/').reverse()[0];
+        return parseInt(filename.substring(0, filename.indexOf('.')));
```

Comment:
always specify the second argument (base `10`) in `parseInt` or be lazy and use  `return +filename.substring(...)` to convert it to a number